### PR TITLE
Fixes #6947

### LINKED ISSUE #6947: Show whether TB has already been discovered
From support mail:
Geocaching.com now gives a status note on the top right of a trackable's page letting us know if we've discovered a TB before. It would be wonderful if that same status could be added when we pull TBs up in cgeo.  

I'd be great if in the same basic place you show "Found it" when we pull up a cache page showed "discovered it" on a trackable. We keep logging the same ones over and over between the 2 of us because we'd otherwise have to link to the web browser to see if we've logged it before or not, & that's a pain.

---
I did not yet see this feature on the webpage...seems to be pretty new.

### PR DESCRIPTION
**Pull request by bjdupuis**
Show the trackable as "retrieved" if the user has previously logged a find.

### CODE REVIEW

**Comment by rsudev** [comment]
Test this please

**Comment by bjdupuis** [comment]
Closing. Will reopen after modifications from feedback is addressed.

**Comment by Lineflyer** [comment]
Hint: You might use "Fix #6947" or "Fixes #6947" in the commit message to allow autoclosure of the issue once the commit is merged

**Comment by SammysHP** [comment]
Just prefix the title of the pull request with "WIP:".

**Comment by bjdupuis** [comment]
Reopening after addressing feedback.

File: main/src/cgeo/geocaching/models/Trackable.java [hunk 1]
```diff
@@ -38,6 +38,9 @@
     private String type = null;
     @Nullable
     private Date released = null;
+    private Date disposition = null;
+    private String dispositionLogGuid;
+    private String dispositionType;
```
Comment (UniQP) [inline main/src/cgeo/geocaching/models/Trackable.java]: I would suggest `logDate`, `logGuid` and `logType` as names (not only here but for the whole PR).

File: main/src/cgeo/geocaching/models/Trackable.java [hunk 2]
```diff
@@ -214,6 +220,34 @@ public void setReleased(@Nullable final Date released) {
...
+    @Nullable
+    public Date getDisposition() {
+        if (disposition != null) {
+            return new Date(disposition.getTime());
+        }
+        return null;
+    }
+
+    public void setDisposition(@Nullable final Date disposition) {
+        this.disposition = disposition == null ? null : new Date(disposition.getTime()); // avoid storing external reference in this object
```
Comment (UniQP) [inline main/src/cgeo/geocaching/models/Trackable.java]:
```suggestion
        // avoid storing external reference in this object
        this.disposition = disposition != null ? new Date(disposition.getTime()) : null;
```

File: main/src/cgeo/geocaching/storage/DataStore.java [hunk 3]
```diff
@@ -2200,7 +2223,7 @@ public static boolean clearSearchedDestinations() {
 
         final Cursor cursor = database.query(
                 dbTableTrackables,
-                new String[]{"_id", "updated", "tbcode", "guid", "title", "owner", "released", "goal", "description"},
+                new String[]{"_id", "updated", "tbcode", "guid", "title", "owner", "released", "goal", "description", "retrieved"},
```
Comment (UniQP) [inline main/src/cgeo/geocaching/storage/DataStore.java]: Shouldn't you add the colums `disposition`, `disposition_type` and `disposition_log_guid`?

Comment (bjdupuis) [reply main/src/cgeo/geocaching/storage/DataStore.java]: Yep, thanks.

File: main/src/cgeo/geocaching/models/Trackable.java (hunk 1 above)
Comment (bjdupuis) [reply main/src/cgeo/geocaching/models/Trackable.java]: Fine idea.

**Comment by bjdupuis** [comment]
Thanks @UniQP, changed to accomodate feedback. Sorry I missed the `DataStore` changes for the fields :/.

File: main/src/cgeo/geocaching/models/Trackable.java [hunk 4]
```diff
@@ -214,6 +220,35 @@ public void setReleased(@Nullable final Date released) {
...
+    public void setLogDate(@Nullable final Date logDate) {
+        // avoid storing external reference in this object
+        this.logDate = logDate != null ? new Date(logDate.getTime()) : null;
+    }
+
+    public String getLogType() {
+        return logType;
+    }
+
+    public void setLogType(String logType) {
```
Comment (rsudev) [inline main/src/cgeo/geocaching/models/Trackable.java]: There is already an enum for Logtypes of Trackables. Did you consider using it and decided against?

**Comment by Lineflyer** [comment]
@rsudev You had comments to this PR. Meanwhile @UniQP approved it. Do you approve or do you think changes are needed?

**Comment by UniQP** [comment]
I implemented the change requested by @rsudev in https://github.com/bjdupuis/cgeo/pull/1.

**Pull request merged**